thunx-autoconfigure/src/main/java/com/contentgrid/thunx/gateway/autoconfigure/OpaProperties.java

```diff
@@ -6,11 +6,14 @@
 @Data
 @ConfigurationProperties(prefix = "opa")
 public class OpaProperties {
-    private OpaServiceProperties service;
+
+    static final String DEFAULT_SERVICE_URL = "http://localhost:8181/";
```

Comment:
Why default to something that is approximately never correct in production?

Instead, the thing that fails could be `@ConditionalOnProperty`?

Reply:
Spring-boot autoconfiguration properties never default to something for production ?
1. Either you default to something that is sane for local development
2. Or you disable it by default and you require configuration

I'm not a proponent of (2) in general, because folloing that pattern leads to very verbose configuration. "Fail because you forgot to set those 37 required parameters!!"

In this particular case, it's reasonable to assume you have to configure it ... so ok, `@ConditionalOnProperty` it is :grimacing:

Reply:
Done